crates/wash-cli/src/call.rs

```diff
@@ -166,100 +173,63 @@ pub struct CallCommand {
     #[clap(flatten)]
     opts: ConnectionOpts,
 
-    /// Optional json file to send as the operation payload
-    #[clap(short, long)]
-    pub data: Option<PathBuf>,
-
-    /// Optional file for saving binary response
-    #[clap(long)]
-    pub save: Option<PathBuf>,
-
-    /// When using json output, display binary as binary('b'), string('s'), or both('2')
-    #[clap(long, default_value = "b")]
-    pub bin: char,
+    /// The unique component identifier of the component to invoke
+    #[clap(name = "component-id")]
+    pub component_id: String,
 
-    /// When invoking a test actor, interpret the response as TestResults
-    #[clap(long)]
-    pub test: bool,
-
-    /// wasmCloud host cluster seed. This cluster seed must match the cluster seed used to
-    /// launch the wasmCloud host in order to pass antiforgery checks made by the host
-    /// This is only optional if a default context is available or a context is provided
-    #[clap(
-        short = 'c',
-        long = "cluster-seed",
-        env = "WASMCLOUD_CLUSTER_SEED",
-        value_parser
-    )]
-    pub cluster_seed: Option<ClusterSeed>,
-
-    /// Public key or OCI reference of actor
-    #[clap(name = "actor-id")]
-    pub actor_id: ModuleId,
-
-    /// Operation to invoke on actor
-    #[clap(name = "operation")]
-    pub operation: String,
-
-    /// Payload to send with operation (in the form of '{"field": "value"}' )
-    #[clap(name = "payload")]
-    pub payload: Vec<String>,
+    /// Fully qualified function to invoke on the actor, e.g. `wasi:cli/run.run`
+    #[clap(name = "function")]
+    pub function: String,
 }
 
 pub async fn handle_call(
     CallCommand {
         opts,
-        data,
-        bin,
-        cluster_seed,
-        actor_id,
-        operation,
-        payload,
+        component_id,
+        function,
         ..
     }: CallCommand,
-) -> Result<Vec<u8>> {
-    debug!(
-        "calling actor with operation: {}, data: {}",
-        &operation,
-        payload.join("")
-    );
-    ensure!(
-        "bs2".contains(bin),
-        "'bin' parameter must be 'b', 's', or '2'"
-    );
-    ensure!(
-        data.is_none() || payload.is_empty(),
-        "you can use either -d/--data or the payload args, but not both."
-    );
-    ensure!(!actor_id.is_empty(), "actor ID may not be empty");
-
-    let payload = if let Some(fname) = data {
-        std::fs::read_to_string(fname)?
-    } else {
-        payload.join("")
+) -> Result<String> {
+    ensure!(!component_id.is_empty(), "component ID may not be empty");
```

Comment:
we can set this as required via `clap` right?

Reply:
We do have it as required, but that doesn't stop someone from supplying an empty string:
```bash
➜ cargo run -p wash-cli -- call "" "wasi:cli/run.run"
    Finished dev [unoptimized + debuginfo] target(s) in 0.44s
     Running `target/debug/wash call '' 'wasi:cli/run.run'`

component ID may not be empty
```

Reply:
Ah so it looks like what you can do is `#[value_parser = "clap::builder::NonEmptyStringValueParser" ]` on the argument and it will check that for you... 

Probably not worth though so resolving this